### PR TITLE
feat(admin): support for HTML in notifications

### PR DESCRIPTION
**Pull request by HejdaJakub**
* Now is possible to create new notification (or edit old notificaion) also in HTML format (not just plain text template).

### CODE REVIEW

**Review by xkureck — CHANGES_REQUESTED** [review]
image
https://uxplanet.org/tabs-for-mobile-ux-design-d4cc4d9410d1
posible solution:
image
https://ux.stackexchange.com/questions/16281/elegant-simple-and-obvious-sub-tabs

image
Also, I and PO think that maybe some sort of information would be nice if they would be placed on place 1 or 2 - information about how this actually works:
- when the HTML notification is filled then this one will be sent to user, the plain notification will be used as a backup for the email clients that do not support HTML in email
- don't forget to change both(HTML + plain) formats when updating some of the formats

**Review by bodnara — CHANGES_REQUESTED** [review]
From a functional point of view other than the lable change I believe there was a requirement for validation of the HTML.

Now for the fun part. The dialog itself would benefit from some refactoring and decomposition as it handles a lot of logic.
Using the "container-presenter" pattern could separate the handling of `vo` and `group` specific actions while reusing the UI. You can look at how the `add-member-dialog` is split.
I would also suggest creating separate components for the language and notification "type" tabs respectively, to "lighten" the dialog itself. I know this type of refactor was not accounted for  but would be a huge improvement at least in my opinion.

From a UI standpoint:
Replace the label: text approach with a placeholders we commonly use (not 100% part of  this issue but a quick fix).
image

Consider the possibility of not having each language as a new tab but having one tab for the content of notifications and changing the language there. It would remove the tabs nesting. Also because when the `tags` section is open it look very clutered.

File: apps/admin-gui/src/app/shared/components/dialogs/add-edit-notification-dialog/add-edit-notification-dialog.component.html [hunk 1]
```diff
@@ -114,48 +114,78 @@ <h1 mat-dialog-title>{{'DIALOGS.NOTIFICATIONS_ADD_EDIT_MAIL.TITLE_EDIT' | transl
...
+              <i [innerHTML]="'DIALOGS.NOTIFICATIONS_ADD_EDIT_MAIL.FORMAT_INFO' | translate"></i>
+            </perun-web-apps-alert>
+            <perun-web-apps-alert alert_type="info" *ngIf="!htmlAuth">
+              {{'DIALOGS.NOTIFICATIONS_ADD_EDIT_MAIL.FORMAT_INFO_NOT_AUTHORIZED' | translate}}
+            </perun-web-apps-alert>
+            <mat-tab-group color="primary">
+              <mat-tab
+                *ngFor="let format of formats"
+                [disabled]="format === 'html' && !htmlAuth"
+                label="{{'DIALOGS.NOTIFICATIONS_ADD_EDIT_MAIL.FORMAT_' + format | uppercase | translate}}">
```
[the remark this answers is not in the document]
Comment (HejdaJakub) [reply apps/admin-gui/src/app/shared/components/dialogs/add-edit-notification-dialog/add-edit-notification-dialog.component.html]: Nice workaround. I changed it.

**Comment by HejdaJakub** [comment]
Functional point: at this moment, we can skip the HTML validation (only Perun Admin has rights for HTML notification settings now).
Other points (UI improvements and refactor): we agreed with @bodnara that other points will be solved in the separate pull-request

**Review by bodnara — APPROVED** [review]

**Pull request merged**